app/presenters/bill-runs/two-part-tariff/charge-reference-details.presenter.js

```diff
@@ -0,0 +1,117 @@
+  const { hasAggregateOrChargeFactor, adjustments } = _adjustments(reviewChargeReference)
+
+  return {
+    billRunId: billRun.id,
+    financialYear: _financialYear(billRun.toFinancialYearEnding),
+    chargePeriod: _prepareDate(
+      reviewChargeReference.reviewChargeVersion.chargePeriodStartDate,
+      reviewChargeReference.reviewChargeVersion.chargePeriodEndDate
+    ),
+    chargeReference: {
+      reference: reviewChargeReference.chargeReference.chargeCategory.reference,
+      description: reviewChargeReference.chargeReference.chargeCategory.shortDescription,
+      totalBillableReturns: _totalBillableReturns(reviewChargeReference.reviewChargeElements),
+      authorisedVolume: reviewChargeReference.chargeReference.volume,
+      adjustments,
+      additionalCharges: _additionalCharges(reviewChargeReference.chargeReference)
+    },
+    licenceId,
+    hasAggregateOrChargeFactor
+  }
+}
+
+function _additionalCharges (chargeReference) {
+  const { supportedSourceName, waterCompanyCharge } = chargeReference
+
+  const charges = []
+
+  if (supportedSourceName) {
+    charges.push(`Supported source ${supportedSourceName}`)
+  }
+
+  if (waterCompanyCharge) {
+    charges.push('Public Water Supply')
+  }
+
+  return charges.join(', ')
+}
+
+function _adjustments (reviewChargeReference) {
+  const adjustments = []
+  let hasAggregateOrChargeFactor = false
+
+  if (reviewChargeReference.amendedAggregate !== 1) {
+    adjustments.push(`Aggregate factor (${reviewChargeReference.amendedAggregate})`)
+    hasAggregateOrChargeFactor = true
+  }
+
+  if (reviewChargeReference.amendedChargeAdjustment !== 1) {
+    adjustments.push(`Charge adjustment (${reviewChargeReference.amendedChargeAdjustment})`)
+    hasAggregateOrChargeFactor = true
+  }
+
+  if (reviewChargeReference.abatementAgreement !== 1) {
+    adjustments.push(`Abatement agreement (${reviewChargeReference.abatementAgreement})`)
+  }
+
+  if (reviewChargeReference.winterDiscount) {
+    adjustments.push('Winter discount')
+  }
+
+  if (reviewChargeReference.twoPartTariffAgreement) {
+    adjustments.push('Two part tariff agreement')
+  }
+
+  if (reviewChargeReference.canalAndRiverTrustAgreement) {
+    adjustments.push('Canal and River trust agreement')
+  }
+
+  return { adjustments, hasAggregateOrChargeFactor }
+}
+
+function _financialYear (financialYearEnding) {
+  const startYear = financialYearEnding - 1
+  const endYear = financialYearEnding
+
+  return `${startYear} to ${endYear}`
+}
+
+function _prepareDate (startDate, endDate) {
+  const preparedStartDate = formatLongDate(startDate)
+  const preparedEndDate = formatLongDate(endDate)
+
+  return `${preparedStartDate} to ${preparedEndDate}`
+}
+
+function _totalBillableReturns (reviewChargeElements) {
+  let totalBillableReturns = 0
+
+  reviewChargeElements.forEach((reviewChargeElement) => {
+    totalBillableReturns += reviewChargeElement.amendedAllocated
+  })
+
+  return totalBillableReturns
```

Comment:
I think this is one of those rare cases where `reduce()` is actually applicable!

```suggestion
  return reviewChargeElements.reduce((total, reviewChargeElement) => {
    total += reviewChargeElement.amendedAllocated

    return total
  }, 0)
```